File: scripts/auto_yes.js
#!/usr/bin/env node
/**
 * Auto-answer 'y' to prompts a spawned command prints (heuristic based).
 * Usage: node scripts/auto_yes.js -- <command> [args...]
 * Example: node scripts/auto_yes.js -- npm init
 */
import { spawn } from 'node:child_process';

// Exported for testability
export const promptRegex = /(?:(?:^|\s)\?\s.*$|\[Y\/n\]|\(y\/n\)|\b(confirm|overwrite|replace|proceed|continue)\b)/i;

export function looksLikePrompt(text) {
  return promptRegex.test(text);
}

function parseArgs(argv = process.argv) {
  const idx = process.argv.indexOf('--');
  if (idx === -1 || idx === process.argv.length - 1) {
    console.error('Usage: node scripts/auto_yes.js -- <command> [args...]');
    process.exit(1);
  }
  return { cmd: process.argv[idx + 1], args: process.argv.slice(idx + 2) };
}

function run() {
  const { cmd, args } = parseArgs();
  console.log(`[auto_yes] Spawning: ${cmd} ${args.join(' ')}`);
  const child = spawn(cmd, args, { stdio: ['pipe', 'pipe', 'pipe'], env: process.env });

  const handle = (chunk, isErr = false) => {
    const text = chunk.toString();
    (isErr ? process.stderr : process.stdout).write(text);
    if (looksLikePrompt(text)) {
      child.stdin.write('y\n');
    }
  };

  child.stdout.on('data', (c) => handle(c, false));
  child.stderr.on('data', (c) => handle(c, true));
  child.on('exit', (code, signal) => {
    console.log(`[auto_yes] Child exited code=${code} signal=${signal}`);
    process.exit(code ?? 0);
  });
}

if (import.meta.url === `file://${process.argv[1]}`) {
  // Executed directly
  run();
}

File: scripts/auto_yes.py
#!/usr/bin/env python3
"""Auto answer 'y' to interactive prompts using pexpect.
Usage:
  python scripts/auto_yes.py <command with args>
Example:
  python scripts/auto_yes.py pip uninstall paquete
Requires: pexpect (pip install pexpect)
"""
import sys
import pexpect

if len(sys.argv) < 2:
    print("Usage: python scripts/auto_yes.py <command> [args...]")
    sys.exit(1)

command = sys.argv[1:]
# Join into a single string for pexpect.
cmd_str = ' '.join(command)
print(f"[auto_yes.py] Spawning: {cmd_str}")

# Patterns to detect a yes/no style prompt.
patterns = [r"\(Y/n\)", r"\[Y/n\]", r"\(y/n\)", r"?", r"overwrite", r"replace", pexpect.EOF, pexpect.TIMEOUT]
child = pexpect.spawn(cmd_str, encoding='utf-8', timeout=60)

while True:
    i = child.expect(patterns)
    if i in (0, 1, 2, 3, 4, 5):  # matched prompt-like text
        try:
            child.sendline('y')
        except Exception:
            break
    elif patterns[i] is pexpect.EOF:  # ended
        break
    else:  # timeout
        # Optionally continue
        pass

child.close()
print(f"[auto_yes.py] Exit status: {child.exitstatus}")
sys.exit(child.exitstatus or 0)
